drivers/kscan/kscan_mchp_xec.c

```diff
@@ -370,45 +448,63 @@ static const struct kscan_driver_api kscan_xec_driver_api = {
 	.enable_callback = kscan_xec_enable_interface,
 };
 
-static int kscan_xec_init(const struct device *dev);
-
-DEVICE_DT_INST_DEFINE(0,
-		    &kscan_xec_init,
-		    NULL,
-		    NULL, NULL,
-		    POST_KERNEL, CONFIG_KSCAN_INIT_PRIORITY,
-		    &kscan_xec_driver_api);
-
-
 static int kscan_xec_init(const struct device *dev)
 {
-	ARG_UNUSED(dev);
+	struct kscan_xec_config const *cfg = dev->config;
+	struct kscan_xec_data *const data = dev->data;
+	struct kscan_regs *regs = cfg->regs;
+
+	kscan_clr_slp_en(dev);
 
 	/* Enable predrive */
-	base->KSO_SEL |= BIT(MCHP_KSCAN_KSO_EN_POS);
-	base->EXT_CTRL = MCHP_KSCAN_EXT_CTRL_PREDRV_EN;
-	base->KSO_SEL &= ~BIT(MCHP_KSCAN_KSO_EN_POS);
-	base->KSI_IEN = MCHP_KSCAN_KSI_IEN_REG_MASK;
+	regs->KSO_SEL |= BIT(MCHP_KSCAN_KSO_EN_POS);
+	regs->EXT_CTRL = MCHP_KSCAN_EXT_CTRL_PREDRV_EN;
+	regs->KSO_SEL &= ~BIT(MCHP_KSCAN_KSO_EN_POS);
+	regs->KSI_IEN = MCHP_KSCAN_KSI_IEN_REG_MASK;
 
 	/* Time figures are transformed from msec to usec */
-	kbd_data.deb_time_press = (uint32_t)
+	data->deb_time_press = (uint32_t)
 		(CONFIG_KSCAN_XEC_DEBOUNCE_DOWN * MSEC_PER_MS);
-	kbd_data.deb_time_rel = (uint32_t)
+	data->deb_time_rel = (uint32_t)
 		(CONFIG_KSCAN_XEC_DEBOUNCE_UP * MSEC_PER_MS);
-	kbd_data.poll_period = (uint32_t)
+	data->poll_period = (uint32_t)
 		(CONFIG_KSCAN_XEC_POLL_PERIOD * MSEC_PER_MS);
-	kbd_data.poll_timeout = 100 * MSEC_PER_MS;
+	data->poll_timeout = 100 * MSEC_PER_MS;
 
-	k_sem_init(&kbd_data.poll_lock, 0, 1);
-	atomic_set(&kbd_data.enable_scan, 1);
+	k_sem_init(&data->poll_lock, 0, 1);
+	atomic_set(&data->enable_scan, 1);
 
-	k_thread_create(&kbd_data.thread, kbd_data.thread_stack,
+	k_thread_create(&data->thread, data->thread_stack,
 			TASK_STACK_SIZE,
-			polling_task, NULL, NULL, NULL,
+			(void (*)(void *, void *, void *))polling_task,
+			(void *)dev, NULL, NULL,
 			K_PRIO_COOP(4), 0, K_NO_WAIT);
 
 	/* Interrupts are enabled in the thread function */
-	IRQ_CONNECT(MCHP_KSAN_NVIC, 0, scan_matrix_xec_isr, NULL, 0);
+	IRQ_CONNECT(DT_INST_IRQN(0), DT_INST_IRQ(0, priority),
+		    scan_matrix_xec_isr, DEVICE_DT_INST_GET(0), 0);
 
 	return 0;
 }
+
+static struct kscan_xec_data kbd_data;
+
+static struct kscan_xec_config kscan_xec_cfg_0 = {
```

Comment:
non-blocking naming suggestion:
```suggestion
static struct kscan_xec_data data;

static struct kscan_xec_config config = {
```

Reply:
skipping this suggestion